scripts/reboot_helper.py

```diff
@@ -0,0 +1,224 @@
+    """
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch to reboot module")
+        return False
+
+    # Attempt to reboot the module
+    if not hasattr(platform_chassis, 'reboot'):
+        log.log_error("Reboot method not found in platform chassis")
+        return False
+
+    if module_name.upper() not in get_all_dpus():
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    log.log_info("Rebooting module {}...".format(module_name))
+    try:
+        platform_chassis.reboot(module_name)
+        log.log_info("Reboot command sent for module {}".format(module_name))
+    except NotImplementedError:
+        raise NotImplementedError("Reboot not implemented on this platform: {type(e).__name__}")
+    except Exception as e:
+        log.log_error("Unexpected error occurred while rebooting module {}: {}".format(module_name, e))
+        return False
+
+    return True
+
+
+def is_dpu():
+    """Check if script is running on DPU module"""
+
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        return False
+
+    # Load platform.json
+    platform_info = device_info.get_platform_info()
+    platform = platform_info.get('platform')
+    if not platform:
+        log.log_error("Platform does not exist in platform_info")
+        return False
+
+    config_data = load_platform_json(platform)
+    if config_data is None:
+        return False
+
+    return any(re.search(r'\.DPU$', key) for key in config_data.keys())
+
+
+def pci_detach_module(module_name):
+    """
+    Detach the specified module by invoking the platform API.
+
+    Args:
+        module_name (str): The name of the module to detach.
+
+    Returns:
+        bool: True if the detach command was successfully sent, False otherwise.
+    """
+
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch to detach module")
+        return False
+
+    # Attempt to detach the module
+    if not hasattr(platform_chassis, 'pci_detach'):
+        log.log_error("PCI detach method not found in platform chassis")
+        return False
+
+    if module_name.upper() not in get_all_dpus():
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    log.log_info("Detaching module {}...".format(module_name))
+    try:
+        platform_chassis.pci_detach(module_name)
+        log.log_info("Detach command sent for module {}".format(module_name))
+        return True
+    except NotImplementedError:
+        raise NotImplementedError("PCI detach not implemented on this platform: {type(e).__name__}")
+    except Exception as e:
+        log.log_error("Unexpected error occurred while detaching module {}: {}".format(module_name, e))
```

Comment:
@vvolam we agreed that pci detach/attach need not be platform specific?

Reply:
@prgeor yes, I am not using attach method but we planned to support detach method in future. But Gagan from NVIDIA mentioned they are doing something special in this PCI detach method and provided comment to modify accordingly.

@gpunathilell @oleksandrivantsiv FYI.

Reply:
@vvolam, @prgeor in the HLD, we defined an API for attaching and detaching the DPU PCI, which was added to the base class in [PR #504](https://github.com/sonic-net/sonic-platform-common/pull/504) of the sonic-platform-common repository. Why aren’t we using it in the implementation? Our agreement was that if the vendor implemented the API, we would use it; otherwise, we would fall back on the default attach/detach implementation.

Reply:
Updated reboot HLD, also updated PR using PCI detach/reattach APIs.